README.md

```diff
@@ -0,0 +1,126 @@
+  - [Transfer token to contract](#Transfer-token-to-contract)
+  - [View functions](#View-functions)
+- [Contract interfaces](#Contract-interfaces)
+  - [Pool creator interfaces](#Pool-creator-interfaces)
+  - [Whitelist admin interfaces](#whitelist-admin-interfaces)
+  - [Types of msg field in ft_transfer_call](#Types-of-msg-field-in-ft_transfer_call)
+    - [AddLiquidity](#AddLiquidity)
+    - [Convert](#Convert)
+
+## Terminology
+
+- `nep141`: [A standard interface for fungible tokens in near network.](https://nomicon.io/Standards/FungibleToken/Core)
+- `conversion pool`: A conversion pool contain a pair of tokens and allows user convert a kind of token to another at a certain rate.
+- `pool creator`: People who create a conversion pool.
+- `rate`: Pool creator can set converting rate when create pool. Then user can convert tokens by this immutable rate in this pool.
+- `whitelist`: Pool creator can only create a conversion pool for tokens in whitelist.
+- `reversible`: By default, the conversion pool is one-way mapping, which means you can only convert token A to B. But when create pool, creator can also select whether users are allowed to convert token reversely, which means the users can exchange token A and token B in both directions.
+- `user`: People who use a conversion pool to convert tokens.
+- `admin`: People who can manage whitelist.
+- `from_token`: If a conversion pool can convert `token A` to `token B`, we use `from_token` refer to `token A`.
+- `to_token`: If a conversion pool can convert `token A` to `token B`, we use `to_token` refer to `token B`.
+
+## Function specification
+
+### Whitelist management
+
+Based on the design of `nep141`, different tokens can use same icon and name, so we need to provide a whitelist to protect users from fraud.
+
+In this contract, the actions that `admin` can perform are as the following:
+
+- Add token into whitelist.
+- Remove token from whitelist.
+
+Refer to [whitelist admin interfaces](#whitelist-admin-interfaces) for the contract interfaces.
+
+### Create conversion pool
+
+Anyone can create a conversion pool for a pair of tokens in whitelist. When someone creates a pool, he needs to set conversion `rate` and whether the pool is `reversible`. The `rate` and `reversible` of the pool can't be updated or deleted after it is created.
+
+Refer to [pool creator interfaces](#pool-creator-interfaces) for the contract interfaces.
+
+### Transfer token to contract
+
+Anyone transfers token to this contract needs to specify purpose. If not, tokens will be fully refunded.
+
+In this contract, the valid purposes are as the following:
+
+- `Adding liquidity` - Anyone can transfer `nep141 token` to this contract for adding liquidity to a pool. We have two rules for checking if transferred token is meaningful:
+  - Transferred token must be `from token` or `to token`.
+  - Transferred token can be `from token` only when `conversion pool` is `reversible`.
```

Comment:
It seems that when pool is set to `reversible`, the `to token` should be allowed here rather than `from token`.

Reply:
Oh,it's my fault,it should be `to token`